distributions/otelcol-contrib-cgo/README.md

```diff
@@ -0,0 +1,11 @@
+# OpenTelemetry Collector Contrib CGO Distro
```

Comment:
As a user, I would find it confusing. Why should I care? What does it bring that others don't? 

I'm not sure I like the idea of a distribution with just a different platform or compilation mechanism. I would prefer a solution where we built with the right options based on what we need to ship.

Does hostmetrics require cgo when compiled for Darwin? Then we should use cgo to compile the distributions that include that component for Darwin.

I find a separate cgo distribution confusing, and opens the door for a cgo distribution for other platforms, which is not something we want.

Reply:
My concern is that when CGO is enabled, the binary is coupled to certain OS versions and will support less OS versions.

Reply:
@jpkrohling 
Hi, I have updated the PR to just enable CGO for Macos otelcol contrib, could you review it again?

Reply:
I would like to have it discussed as part of the SIG and have buy-in from other maintainers.

Reply:
Hi, I have tested with golang 10.15 and it works, which is also the oldest macos version that golang >= 1.21 supported [ref](https://go.dev/wiki/MinimumRequirements)
I think we could just turn on the CGO option without a separated distribution.